tests/device_tests/zcash/test_sign_tx.py

```diff
@@ -0,0 +1,256 @@
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib import btc, messages
+from trezorlib.exceptions import TrezorFailure
+from trezorlib.tools import parse_path
+
+from ..bitcoin.signtx import (
+    request_finished,
+    request_input,
+    request_output,
+)
+
+B = messages.ButtonRequestType
+
+TXHASH_aaf51e = bytes.fromhex(
+    "aaf51e4606c264e47e5c42c958fe4cf1539c5172684721e38e69f4ef634d75dc"
+)
+TXHASH_e38206 = bytes.fromhex(
+    "e3820602226974b1dd87b7113cc8aea8c63e5ae29293991e7bfa80c126930368"
+)
+
+pytestmark = [pytest.mark.altcoin, pytest.mark.zcash]
+
+@pytest.mark.skip_t1
+def test_version_group_id_missing(client):
+    inp1 = messages.TxInputType(
+        # tmQoJ3PTXgQLaRRZZYT6xk8XtjRbr2kCqwu
+        address_n=parse_path("m/44h/1h/0h/0/0"),
+        amount=300000000,
+        prev_hash=TXHASH_e38206,
+        prev_index=0,
+    )
+    out1 = messages.TxOutputType(
+        address="tmJ1xYxP8XNTtCoDgvdmQPSrxh5qZJgy65Z",
+        amount=300000000 - 1940,
+        script_type=messages.OutputScriptType.PAYTOADDRESS,
+    )
+
+    with pytest.raises(TrezorFailure, match="Version group ID must be set."):
+        btc.sign_tx(
+            client,
+            "Zcash Testnet",
+            [inp1],
+            [out1],
+            version=5,
+        )
+
+@pytest.mark.skip_t1
+def test_one_one(client):
+    inp1 = messages.TxInputType(
+        # tmQoJ3PTXgQLaRRZZYT6xk8XtjRbr2kCqwu
+        address_n=parse_path("m/44h/1h/0h/0/0"),
+        amount=300000000,
+        prev_hash=TXHASH_e38206,
+        prev_index=0,
+    )
+
+    out1 = messages.TxOutputType(
+        address="tmJ1xYxP8XNTtCoDgvdmQPSrxh5qZJgy65Z",
+        amount=300000000 - 1940,
+        script_type=messages.OutputScriptType.PAYTOADDRESS,
+    )
+
+    with client:
+        client.set_expected_responses(
+            [
+                request_input(0),
+                request_output(0),
+                messages.ButtonRequest(code=B.ConfirmOutput),
+                messages.ButtonRequest(code=B.SignTx),
+                request_input(0),
+                request_output(0),
+                request_finished(),
+            ]
+        )
+
+        _, serialized_tx = btc.sign_tx(
+            client,
+            "Zcash Testnet",
+            [inp1],
+            [out1],
+            version=5,
+            version_group_id=0x892F2085,
+            branch_id=0x76B809BB,
+        )
+
+        # TODO: send tx to testnet
+        # TODO: check serialized_tx
```

Comment:
Let's be sure that the transactions work on testnet, before merging the code.

Reply:
I agree. I will submit these transactions as soon as the testnet will be relaunched, which should be in the beginning of April.